alphazero/mcts/mcts.py

```diff
@@ -0,0 +1,282 @@
+from __future__ import annotations
+from typing import List, TypeVar
+import numpy as np
+import torch
+from alphazero import TicTacToe
+from alphazero.utils import sample_tensor_indices
+
+GameClassType = TypeVar("T", bound="Game")
+
+
+class BoardConverter:
+    @staticmethod
+    def board_to_tensor(board):
+        raise NotImplementedError
+
+    @staticmethod
+    def action_to_tensor(action):
+        # Not sure if needed
+        raise NotImplementedError
+
+    @staticmethod
+    def tensor_to_action(action_tensor):
+        raise NotImplementedError
+
+
+class TicTacToeConverter(BoardConverter):
+    @staticmethod
+    def board_to_tensor(board):
+        player_1 = board == 0
+        player_2 = board == 1
+        player = 1 if player_1.sum() > player_2.sum() else 0
+        player_layer = torch.full(board.size(), player)
+        return torch.stack((player_1, player_2, player))
+
+    ## TODO: Flesh this out and move
```

Comment:
guessing this means moving this to a different file? Since we're eventually going to use it for the network, maybe it should go under utils?

Reply:
Stashed for now